File: stock-serveur-v2/src/main/java/cm/belrose/stockserveur/model/LigneVente.java
package cm.belrose.stockserveur.model;

import cm.belrose.stockserveur.config.audit.Auditable;
import lombok.*;
import org.hibernate.envers.NotAudited;

import javax.persistence.*;
import java.io.Serializable;
import java.math.BigDecimal;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
@Entity
@Table(name = "ligneVentes")
public class LigneVente extends Auditable<String> implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private BigDecimal quantite;
    private BigDecimal prixUniataire;
    private Long entrepriseId; //Ce champs est juste mis pour simplifier les choses
    @NotAudited
    @ManyToOne
    private Article article;
    @NotAudited
    @ManyToOne
    private Vente vente;
}

File: stock-client-v2/src/gs-api/src/models/ligne-vente.ts
/* tslint:disable */
import { Article } from './article';
import { Vente } from './vente';
export interface LigneVente {
  createdBy?: string;
  updatedBy?: string;
  createdDate?: string;
  updateDate?: string;
  id?: number;
  quantite?: number;
  prixUniataire?: number;
  entrepriseId?: number;
  article?: Article;
  vente?: Vente;
}
